runtimes/peregrine/src/lib.rs

```diff
@@ -778,15 +778,23 @@ construct_runtime! {
 
 impl did::DeriveDidCallAuthorizationVerificationKeyRelationship for Call {
```

Comment:
C: I would move this logic (and perhaps anything else that is in common between spiritnet and peregrine) into `primitives`, so that we avoid duplicating code. What do you think?

Reply:
would love that! But we implement that for Call which is an enum defined in the runtime

Reply:
Mhh yeah ok, and also the runtime itself is an enum so not very easy to extend with runtime-specific logic if we wanted to 😕